Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -315,11 +315,9 @@ const struct xdg_toplevel_listener xdgToplevelListener = {
         }
 
         uint32_t pendingState = 0;
-        const auto* stateData = static_cast<uint32_t*>(states->data);
-        for (size_t i = 0; i < states->size; i++) {
-            uint32_t state = stateData[i];
-
-            switch (state) {
+        const char* end = static_cast<const char*>(states->data) + states->size;
+        for (uint32_t* state = static_cast<uint32_t*>(states->data); reinterpret_cast<const char*>(state) < end; ++state) {
```

Comment:
I usually employ 'auto' on the LHS, if RHS contains an explicit cast, to avoid typing out the type twice.
I'd also change 'end' to be of the same type as 'state', to avoid the reinterpret_cast in the loop condition.